src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/RejectionConnection.cs

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Infrastructure;
+using Microsoft.AspNetCore.Server.Kestrel.Internal.System.IO.Pipelines;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal
+{
+    public class RejectionConnection : IConnectionContext
+    {
```

Comment:
We've gone in circles on this. At this point, this is the lowest cost implementation that avoids issues such as throttling TLS handshake, setting timeouts, etc.

I'll open an issue to add the ability to send a 503 response. I've kept my partial implementation of that so we can revisit this post 2.0.